src/components/ScrollGradient/__tests__/ScrollGradient.spec.js

```diff
@@ -58,22 +60,20 @@ describe('ScrollGradient', () => {
       expect(onScroll).toHaveBeenCalledTimes(1);
     });
 
-    it('adds resize event listener when mounted', () => {
-      const spy = jest.spyOn(window, 'addEventListener');
+    it('adds `ResizeObserver` when mounted', () => {
       scrollGradient = mount(
         <ScrollGradient className={className} color="blue">
           {children}
         </ScrollGradient>
       );
-      expect(spy).toHaveBeenCalledWith('resize', expect.any(Function));
-      spy.mockRestore();
+
+      expect(resizeObserverMock.mock.instances[6].observe).toBeCalled();
```

Comment:
With my previous comment, you would be able to write your assertion to something like:

```js
      expect(observeMock).toBeCalledTimes(1);
      // or
      expect(observeMock).toBeCalledWith(1);
```